### ISSUE TITLE
BUG: Axis mismatch in experiment 6 boundary concatenation

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#49]
## Description

In `experiments/experiment_6/train.py`, line 193:

```python
bc_left_wall = jnp.concatenate([bc_left_wall_bottom, bc_left_wall_above], axis=1)
```

Both inputs have shape `(num_batches, batch_size, 3)`. Concatenating along `axis=1` produces `(num_batches, 2*batch_size, 3)`, but downstream loss functions expect uniform `batch_size` across all boundary terms.

## Impact

- Left wall batches have double the points compared to other walls
- Implicit over-weighting of left wall BC losses
- Potential shape errors inside `lax.scan`

## Fix

Either:
- (a) Halve `n_bc_per_wall` for left-wall sub-segments so the concatenated result matches `batch_size`
- (b) Concatenate along `axis=0` to double the number of batches
- (c) Interleave sampling within `generate_epoch_data`

## Category
**Critical Bug** · Pillar: Logic Check
